sanic/router.py

```diff
@@ -161,10 +163,10 @@ def add_parameter(match):
             parameters.append(parameter)
 
             # Mark the whole route as unhashable if it has the hash key in it
-            if re.search('(^|[^^]){1}/', pattern):
+            if re.search(r'(^|[^^]){1}/', pattern):
                 properties['unhashable'] = True
             # Mark the route as unhashable if it matches the hash key
-            elif re.search(pattern, '/'):
```

Comment:
It was plain wrong. 

Note that the above comment says `Mark the route as unhashable if it matches the hash key`, the hash key is `/`, so `re.search(pattern, '/'):` means search `pattern` in `/` not search `/` in `pattern`.